domain/permission/service/service.go

```diff
@@ -17,15 +18,17 @@ type State interface {
 	// CreatePermission gives the user access per the provided spec.
 	// It requires the user/target combination has not already been
 	// created.
-	CreatePermission(ctx context.Context, spec UserAccessSpec) (permission.UserAccess, error)
+	CreatePermission(ctx context.Context, spec state.UserAccessSpec) (permission.UserAccess, error)
```

Comment:
FYI: we have a rule that service doesn't know about the state package and the service package doesn't know about state. So in this case you could have a file called userspec.go in the root of the `domain/permission` package, that way they don't depend on each other. Alternatively, you could move it to core if it has more use outside of the domain.

Reply:
done